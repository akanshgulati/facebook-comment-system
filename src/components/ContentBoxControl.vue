<template>
  <div class="control-box">
    <Link :value="depth === 1 ? 'Comment': 'Reply'" @click="toggle" type="primary"/>
    <Link :value="secondaryText" @click="toggle" type="secondary"/>
  </div>
</template>
<script>
import { getPlural } from '../utils/CommonUtils';

import Link from './Link';

export default {
  methods: {
    toggle() {
      this.$emit('toggleChild');
    },
    getPlural,
  },
  props: {
    depth: {
      type: Number,
      required: true,
    },
    count: {
      type: Number,
      required: true,
    },
  },
  components: {
    Link,
  },
  computed: {
    secondaryText() {
      // for depth > 2 we are not storing nested replies
      if (this.depth > 2) {
        return '';
      }
      return `${this.count} ${this.depth === 1 ? getPlural('Comment', 'Comments', this.count) : getPlural('Reply',
        'Replies', this.count)}`;
    },
  },
};
</script>
<style>

  .control-box {
    display: flex;
    justify-content: space-between;
  }

  a.control-box__link-left{
    color: #ff7133;
  }

  a.control-box__link-left:visited {
    color: #ff7133;
  }

  .control-box__link-right {
    margin-left: auto;
  }
</style>
