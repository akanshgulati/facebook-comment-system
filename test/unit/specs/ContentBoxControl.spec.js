import Vue from 'vue';
import ContentBoxControl from '@/components/ContentBoxControl';
import { createRenderer } from 'vue-server-renderer';

describe('ControlBoxControl.vue', () => {
  it('should render correct contents', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 1,
        count: 2,
      },
    }).$mount();
    expect(vm.$el.children.length).toEqual(2);
  });
  it('should show comment string for depth 1', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 1,
        count: 2,
      },
    }).$mount();
    expect(vm.$el.children[0].innerHTML).toContain('Comment');
  });
  it('should show reply string for depth 2', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 2,
        count: 2,
      },
    }).$mount();
    expect(vm.$el.children[0].innerHTML).toContain('Reply');
  });
  it('should show replies count for count 2', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 2,
        count: 2,
      },
    }).$mount();
    expect(vm.$el.children[1].innerHTML).toContain('2');
    expect(vm.$el.children[1].innerHTML).toContain('Replies');
  });
  it('should show reply count for count 1 and depth 2', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 2,
        count: 1,
      },
    }).$mount();
    expect(vm.$el.children[1].innerHTML).toContain('1');
    expect(vm.$el.children[1].innerHTML).toContain('Reply');
  });
  it('should show reply count for count 1 and depth 1', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 1,
        count: 1,
      },
    }).$mount();
    expect(vm.$el.children[1].innerHTML).toContain('1');
    expect(vm.$el.children[1].innerHTML).toContain('Comment');
  });
  it('should show reply count for count 2 and depth 1', () => {
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 1,
        count: 2,
      },
    }).$mount();
    expect(vm.$el.children[1].innerHTML).toContain('2');
    expect(vm.$el.children[1].innerHTML).toContain('Comments');
  });

  it('should match the snapshot with depth 1 and count 1', (done) => {
    const renderer = createRenderer();
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 1,
        count: 1,
      },
    }).$mount();
    renderer.renderToString(vm, (err, str) => {
      expect(str).toMatchSnapshot();
      done();
    });
  });

  it('should match the snapshot with depth 2 and count 2', (done) => {
    const renderer = createRenderer();
    const Constructor = Vue.extend(ContentBoxControl);
    const vm = new Constructor({
      propsData: {
        depth: 2,
        count: 2,
      },
    }).$mount();
    renderer.renderToString(vm, (err, str) => {
      expect(str).toMatchSnapshot();
      done();
    });
  });
});
