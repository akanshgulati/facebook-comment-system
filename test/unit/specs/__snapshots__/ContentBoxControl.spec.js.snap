// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlBoxControl.vue should match the snapshot with depth 1 and count 1 1`] = `
<div class="control-box"><a class="primary">
        Comment
    </a> <a class="secondary">
        1 Comment
    </a></div>
`;

exports[`ControlBoxControl.vue should match the snapshot with depth 2 and count 2 1`] = `
<div class="control-box"><a class="primary">
        Reply
    </a> <a class="secondary">
        2 Replies
    </a></div>
`;
